packages/ui/components/navigation/tabs/navigation.test.tsx

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable playwright/missing-playwright-await */
+import { fireEvent, render, screen } from "@testing-library/react";
+import { PlusIcon } from "lucide-react";
+import { vi } from "vitest";
+
+import HorizontalTabs from "./HorizontalTabs";
+import VerticalTabs from "./VerticalTabs";
+
+vi.mock("@calcom/lib/hooks/useUrlMatchesCurrentUrl", () => ({
+  useUrlMatchesCurrentUrl() {
+    return {
+      route: "/",
+      pathname: "",
+      query: "",
+      asPath: "",
+      push: vi.fn(),
+      events: {
+        on: vi.fn(),
+        off: vi.fn(),
+      },
+      beforePopState: vi.fn(() => null),
+      prefetch: vi.fn(() => null),
+    };
+  },
+}));
+
+vi.mock("@calcom/lib/hooks/useLocale", () => ({
+  useLocale: () => ({ t: (key: string) => key, isLocaleReady: true }),
+}));
+
+describe("Tests for navigation folder", () => {
+  describe("Test HorizontalTabs Component", () => {
+    const mockTabs = [
+      { name: "Tab 1", href: "/tab1" },
+      { name: "Tab 2", href: "/tab2", avatar: "Avatar" },
+      { name: "Tab 3", href: "/tab3" },
+    ];
+
+    beforeEach(() => {
+      vi.clearAllMocks();
+    });
+
+    test("Should render tabs with correct name and href", () => {
+      render(<HorizontalTabs tabs={mockTabs} />);
+      mockTabs.forEach((tab) => {
+        const tabLabelElement = screen.getByTestId(`horizontal-tab-${tab.name}`);
+        expect(tabLabelElement).toBeInTheDocument();
+
+        const name = screen.getByText(tab.name);
+        expect(name).toBeInTheDocument();
+        expect(tabLabelElement).toHaveAttribute("href", tab.href);
+
+        const spanSibling = name.querySelector("span");
```

Comment:
`span` seems to be implementation detail. User doesn't see a span, he would see the content in it.
We should check what that span shows to user.

Reply:
Hi @hariombalhara That span tag doesn't have any content, please take a look here:
![image](https://github.com/calcom/cal.com/assets/121884634/fe3113f7-58a9-44df-86fc-b25422f6eb10)

Reply:
@gitstart-calcom Then why are we checking for `span` tag. What I meant was that we should test for what a user sees. If he can't see the span(it has no content), then why are we testing its presence

Reply:
Makes sense, we have updated the test now